addons/payment/models/payment_acquirer.py

```diff
@@ -234,6 +234,15 @@ def write(self, values):
         self._check_required_if_provider()
         return result
 
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_master_data(self):
+        """ prevent the deletion of the payment acquirer if it reference is used in another module. """
+        external_ids = self.get_external_id()
+        for acquirer in self:
+            external_id = external_ids[acquirer.id]
+            if external_id:
```

Comment:
you should not skip this part https://github.com/odoo/enterprise/commit/39d34f928c147e31a4157a2eb6283c7114de7a4c
```py
            if external_id and not external_id.startswith('__export__'):
```
Manually created record may have external id if it was exported